src/software/new_geom/convex_polygon.cpp

```diff
@@ -0,0 +1,29 @@
+#include "software/new_geom/convex_polygon.h"
+
+ConvexPolygon::ConvexPolygon(const std::initializer_list<Point>& points) : Polygon(points)
+{
+}
+
+double ConvexPolygon::area() const
```

Comment:
Can we add unit tests for this? Currently there a a few tests for rectangle area, but we want to test this for arbitrary Convex polygons

Reply:
ConvexPolygon is an abstract class so we cant actually construct arbitrary polygons, this is meant to be used for rectangle, triangle, and any other convex polygons we implement in the future. Is this okay or did we wanna go another way with this?

Reply:
Ah I see. I think we would want to be able to create arbitrary (convex) polygons, since that is/was what our obstacle system used for navigation. I don't remember if we made an explicit decision about being able to construct arbitrary polygons in our big geom meeting a while ago.

@garethellis0 @jonathanlew @jsn0716 for opinions?

Reply:
I don't recall the exact details of what we discussed in the original architecture meeting, but it's entirely reasonable for both `Polygon` and `ConvexPolygon` to be concrete classes. Certainly we can implement `area()` for the `ConvexPolygon` class, even if it is an abstract class. 

TL;DR: Both `Polygon` and `ConvexPolygon` should be concrete classes, `area` should be implemented for `ConvexPolygon`.

Reply:
```ConvexPolygon``` is now a concrete class in the latest update